### PR TITLE
SPs for inserting/retrieving GridInfo entries via dcid

### PR DESCRIPTION
**Pull request by rjgildea**
Add `upsert_dc_grid` and ``retrieve_grid_info_for_dc` stored procedures for inserting/retrieving `GridInfo` entries via a `dataCollectionId` rather than `dataCollectionGroupId`.

### CODE REVIEW

**Review by KarlLevik — CHANGES_REQUESTED** [review]
I think that looks good, but could you please add separate files for the individual stored procedures under `schemas/ispyb/stored_programs/`? Also, I don't see any grants for the `retrieve_grid_info_for_dc` stored procedure?

File: schemas/ispyb/stored_programs/sp_retrieve_grid_info_for_dc.sql [hunk 1]
```diff
@@ -0,0 +1,27 @@
+DELIMITER ;;
+CREATE PROCEDURE `retrieve_grid_info_for_dcg`(IN p_dcgId int unsigned)
```
Comment (KarlLevik) [inline schemas/ispyb/stored_programs/sp_retrieve_grid_info_for_dc.sql:2]: Looks like the wrong procedure, but you seem to have the right one in `routines.sql`.

**Review by KarlLevik — CHANGES_REQUESTED** [review]
Looks very good, except for the wrong sproc in the "retrieve" sproc file as per comment below.

**Comment by rjgildea** [comment]
> Looks very good, except for the wrong sproc in the "retrieve" sproc file as per comment below.

Good spot - I think the correct one should be in there now!

**Review by KarlLevik — APPROVED** [review]

**Pull request merged**